### ISSUE TITLE
Confusion Matrix for Classification et. al. models.

### ISSUE THREAD

**Issue by codeninja** [ThilinaRajapakse/simpletransformers#637]
**Is your feature request related to a problem? Please describe.**
When training and evaluating a classification model accuracy is not a sufficient metric especially in multi-label classification tasks. Often a 99% accuracy can mis-represent the classification of false positives. The solution is to present the classifications in the form of a confusion matrix.

**Describe the solution you'd like**
After training a classification model I want to see a confusion matrix which graphically represents the Recall and Precision of the model.  While binary and multi-label classification tasks are different in their representation, the implementation is simple and straight forward.

**Additional context**
Here I have included the source for a confusion matrix I have implemented for one of my projects in a multi-label classification task. It's important for our solution that we accurately identify "Leads" against "not leads". We further classify a subclass of the call type (support, enquiry, etc) and while a mis-classification within the primary classification of Lead is acceptable, mis-identifying a "Not lead" transcript as a "Lead" is detrimental.

Here in the CM, we can see that there's a distinct boundary between Leads and Not lead calls, even if there is some misclassified calls within the sub-category. This matrix has assisted us greatly in evaluating the model and it's performance and I can see a use case to include it as an evaluation method.

Note that the code pasted below is specific to our implementation and includes calls to outside classes so it's not expected that this would run for you without error. But it could serve as an example for implementation.

```
from sklearn import metrics
from sklearn.metrics import classification_report
import seaborn as sns
import math

def build_class_weights(classes):
    counts = Counter(classes)
    total = len(classes)
    weights = [math.log(total/max(v, 1)) for k, v in enumerate(counts)]
    return weights

# dp=DataProcessor
def graphCM(y_true, y_pred, dp):
    counts = Counter(y_true)
    labels = [dp.to_label(p) for p in counts.keys()]
    
    fig, (true, pred) = plt.subplots(1,2)
    fig.suptitle('Call Prediction')
    fig.set_figheight(5)
    fig.set_figwidth(18)

    cmat_true = metrics.confusion_matrix(y_true, y_pred, normalize='true', labels=range(0, len(labels)))
    cmat_pred = metrics.confusion_matrix(y_true, y_pred, normalize='pred', labels=range(0, len(labels)))

    sns.heatmap(cmat_true, robust=True, cmap="YlGnBu", annot=True, xticklabels=labels, yticklabels=labels, ax = true, square=True)
    sns.heatmap(cmat_pred, robust=True, cmap="YlGnBu", annot=True, xticklabels=labels, yticklabels=labels, ax = pred, square=True)

    true.set_title(f"Recall")
    pred.set_title(f"Precision")
    plt.show()
    
    cr = classification_report(y_true, y_pred, target_names=labels, labels=range(0, len(labels)), zero_division=0)
    print(cr)


```

Output:
![image](https://user-images.githubusercontent.com/14914/89659724-e0b66d00-d895-11ea-9c29-7b8a6cd151d5.png)

**Comment by ThilinaRajapakse** [comment]
The [default metric ](https://simpletransformers.ai/docs/classification-models/#evaluating-a-classification-model)for multilabel classification is LRAP score, and not accuracy. This is certainly not going to be a useful metric for every use case. The idea was to provide a general case default metric (usually chosen by looking at the metrics used in benchmarks like GLUE) and leave it up to the user to [calculate additional metrics](https://simpletransformers.ai/docs/usage/#additional-evaluation-metrics) by defining custom metric functions which can be passed to `train_model()` and `eval_model()`.

However, I do agree that a confusion matrix would be quite useful for multilabel classification (already implemented for binary and multiclass, albeit through W&B). I'll label this as a feature request, but I'm not sure when I'll be able to look into implementing it.
